File: src/main/java/com/spring/principle/examples/roulette/Roulette.java
package com.spring.principle.examples.roulette;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
@AllArgsConstructor
public class Roulette {
    private Integer score;
    private BigDecimal probability;
    private Integer stocks;

    public void decreaseStock() {
        if (stocks > 0) {
            stocks -= 1;
        }
    }
}

File: src/main/java/com/spring/principle/examples/roulette/Roulette.kt
package com.spring.principle.examples.roulette

import java.math.BigDecimal
import java.util.concurrent.atomic.AtomicInteger


class Roulette(val score: Int, valueOf: BigDecimal, var stocks: AtomicInteger) {
    val probability: BigDecimal = valueOf
    fun decreaseStock() {
        if (stocks.get() > 0) {
            stocks.set(stocks.get() - 1)
        }
    }
}
